src/beignet/_bisect.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from typing import Callable, Literal
+
+import torch
+from torch import Tensor
+
+from beignet._root_scalar import RootSolutionInfo
+
+
+def bisect(
+    f: Callable,
+    *args,
+    lower: float,
+    upper: float,
+    rtol: float | None = None,
```

Comment:
Use the standard tolerance, i.e., `atol=2e-12`.

Reply:
The default tolerance should depend on the dtype so I think it makes more sense for the default to be `None`. Same as what is done in `torch.testing`